clients/src/main/java/org/apache/kafka/clients/producer/Producer.java

```diff
@@ -33,19 +35,80 @@
  * @see MockProducer
  */
 public interface Producer<K, V> extends Closeable {
+    public static final int INVALID_PID = Integer.MAX_VALUE;
+
+    /**
+     * Needs to be called before any of the other transaction methods. Assumes that
+     * the producer.app.id is specified in the producer configuration.
+     *
+     * This method does the following:
+     *   1. Ensures any transactions initiated by previous instances of the producer
+     *      are committed or rolled back.
+     *   2. Gets the internal producer id and epoch, used in all future transactional
+     *      messages issued by the producer.
+     *
+     * @throws IllegalStateException if the appId for the producer is not set
+     *         in the configuration.
+     */
+    void initTransactions() throws IllegalStateException;
```

Comment:
nit: these are all runtime exceptions, so there's no need to explicitly add them to the signature. Not that it hurts, but we don't do it elsewhere, so maybe the javadoc is enough.